.github/workflows/gradle.yml

```diff
@@ -1,27 +1,51 @@
-name: Build with Gradle
+name: Build Gradle
 
-on: [ push, pull_request ]
 
-permissions:
-  contents: read
+on:
+  push:
+  pull_request:
+
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+
+    runs-on: ubuntu-22.04
 
     steps:
-    - uses: actions/checkout@v3
+    # https://github.com/actions/checkout
+    - name: Check out Git repositor
+      uses: actions/checkout@v4.1.7
+
+    # https://github.com/gradle/gradle-build-action
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@v3.5.0
+
+    # https://github.com/gradle/wrapper-validation-action
+    - name: Wrapper validation
+      uses: gradle/actions/wrapper-validation@v3.5.0
+
+    # https://github.com/actions/setup-java
     - name: Set up JDK 8
-      uses: actions/setup-java@v3
+      uses: actions/setup-java@v4.2.1
       with:
         distribution: 'temurin'
         java-version: '8'
-    - name: Build with Gradle
-      uses: gradle/gradle-build-action@67421db6bd0bf253fb4bd25b31ebb98943c375e1
-      with:
-        arguments: shadowJar
-    - name: Upload a Build Artifact
-      uses: actions/upload-artifact@v3.1.0
+        java-package: jdk
+        architecture: x64
+        cache: gradle
+
+    - name: Configure Git
+      run: git config --global user.name "Github Actions" && git config --global user.email "actions@github.com"
+
+    - name: Make Gradlew executable
+      run: chmod +x ./gradlew
+
```

Comment:
```suggestion
```
I think I'd rather not have this, that way if I do ever accidentally break the permissions on `gradlew` I immediately know and can fix it for everyone who clones the repository